src/shared/lrc/index.rs

```diff
@@ -175,57 +186,43 @@ impl LrcIndex {
 }
 
 impl LrcIndexEntry {
-    fn read(read: impl BufRead, path: PathBuf) -> Result<Option<Self>> {
-        let mut title = None;
-        let mut artist = None;
-        let mut album = None;
-        let mut length = None;
-
-        for buf in read.lines() {
-            let buf = buf?;
-            if buf.trim().is_empty() || buf.starts_with('#') {
-                continue;
-            }
+    fn read(mut read: impl BufRead, path: PathBuf) -> Result<Option<Self>> {
+        let mut content = String::new();
 
-            let (metadata, rest) = buf
-                .trim()
-                .strip_prefix('[')
-                .and_then(|s| s.rsplit_once(']'))
-                .with_context(|| format!("Invalid lrc line format: '{buf}'"))?;
-            if !rest.is_empty() {
-                break;
+        loop {
+            let mut line = String::new();
+            if read.read_line(&mut line)? == 0 {
+                break; // EOF
             }
-
-            match metadata.chars().next() {
-                Some(c) if c.is_numeric() => {
-                    break;
-                }
-                Some(_) => {
-                    let (key, value) = metadata
-                        .split_once(':')
-                        .with_context(|| format!("Invalid metadata line: '{metadata}'"))?;
-                    match key.trim() {
-                        "ti" => title = Some(value.trim().to_string()),
-                        "ar" => artist = Some(value.trim().to_string()),
-                        "al" => album = Some(value.trim().to_string()),
-                        "length" => length = Some(parse_length(value.trim())?),
-                        _ => {}
+            // if this line has a timestamp, stop reading
+            // We are looking for lines that start with [ and have a timestamp in them
+            // reading all the way to the end of the file is not necessary
+            let trimmed = line.trim();
+            if !trimmed.is_empty() && !trimmed.starts_with('#') && trimmed.starts_with('[') {
```

Comment:
```suggestion
            if !trimmed.is_empty() && trimmed.starts_with('[') 
```
This check is redundant right?

Reply:
yes, i forgot to change after refractoring the code